src/PolyfillTrait6.php

```diff
@@ -29,10 +29,28 @@ public function expectExceptionMessageMatches($regexp)
     {
         if (\is_callable(['parent', 'expectExceptionMessageMatches'])) {
             parent::expectExceptionMessageMatches($regexp);
-        } elseif (\is_callable(['parent', 'setExpectedExceptionRegExp'])) {
-            $this->setExpectedExceptionRegExp(\Exception::class, $regexp);
-        } else {
+
+            return;
+        }
+
+        // In some PHPUnit versions just setting an expectation for specific
+        // expection message won't trigger exception handler. Therefore we need
+        // to set the expected type, but trying to keep the type.
+        if (\is_callable(['parent', 'getExpectedException'])) {
```

Comment:
Should we try to find out exact versions where this happens?

Reply:
i believe the best attempt is to check the existing features (like method existence) rather than version number, good job here :+1: